### PR TITLE
fix: support for remote development environments [ROAD-1236]

### PR DESCRIPTION
**Pull request by j-luong**
### Description

_This PR fixes support for remote development environments_

### Checklist

- [x] Tests added and all succeed
- [x] Linted
- [x] CHANGELOG.md updated

### CODE REVIEW

**Comment by bastiandoetsch** [comment]
According to [this](https://code.visualstudio.com/api/advanced-topics/extension-host), the change makes vscode run our extension on the workspace (e.g. remote). So if I understand it correctly, it would use the remote token and download to the remove workspace?

**Review by bastiandoetsch — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,10 +1,21 @@
 # Snyk Security - Code and Open Source Dependencies Changelog
-## [1.12.2]
+
+## [1.12.5]
```
Comment (michelkaporin) [inline CHANGELOG.md:3]: I think it's 1.12.3 that's coming up, if you keep commit as "fix"

File: package.json [hunk 2]
```diff
@@ -470,5 +470,8 @@
     "uuid": "^8.3.2",
     "validate-npm-package-name": "^3.0.0",
     "vscode-languageclient": "8.0.0-next.2"
-  }
+  },
+  "extensionKind": [
```
Comment (michelkaporin) [inline package.json:474]: What is the default value VS Code sets this to, when it's unspecified as we had it previously?

**Comment by j-luong** [comment]
> According to [this](https://code.visualstudio.com/api/advanced-topics/extension-host), the change makes vscode run our extension on the workspace (e.g. remote). So if I understand it correctly, it would use the remote token and download to the remove workspace?

@bastiandoetsch yes, but for the remote workspace only. [The change means it will run in the same machine where the workspace is](https://code.visualstudio.com/api/advanced-topics/remote-extensions#architecture-and-extension-kinds); so in the case of a remote workspace, it'll run on the remote workspace's machine.

File: package.json (hunk 2 above)
Comment (j-luong) [reply package.json:474]: [I believe it tries to infer the location automatically](https://code.visualstudio.com/api/advanced-topics/remote-extensions#incorrect-execution-location). One of the [common problems](https://code.visualstudio.com/api/advanced-topics/remote-extensions#incorrect-execution-location) is incorrect execution location; for which setting the `extensionKind` property is a suggested fix.

**Comment by j-luong** [comment]
This PR won't fix the issue.

The issue is to do with the fact that we read from VSCode's secret storage ([which will always be local](https://github.com/microsoft/vscode/blob/dcc6d8f54e620d8841ce36a2cf92d1d03c2a134f/src/vscode-dts/vscode.d.ts#L7121-L7125)) and then authenticate using the CLI (via the Language Server). The CLI writes the API token to the config store as part of the auth flow.

See comment in [HEAD-16](https://snyksec.atlassian.net/browse/HEAD-16?focusedCommentId=630238) for a summary.

**Pull request closed without merging**